### PR TITLE
add styled react components to ecomm

### PR DESCRIPTION
**Pull request by govindappaarun**
As part of this pull request I am converting most of  the CSS components into styled-components
 
 Things to review
    - Components folder  and its sub folders
    
   Ignore
    - backend and  assets folder

### CODE REVIEW

File: src/components/Icon/Icon.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import styled from "styled-components";
+import { ColorTypes, SizeTypes } from "../types/types";
+import { StyleSvgProps } from "./Icon.styled";
+
+import { ReactComponent as Wish } from "../../assets/svg/heart.svg";
+import { ReactComponent as Cart } from "../../assets/svg/shopping-cart.svg";
+import { ReactComponent as User } from "../../assets/svg/user.svg";
+import { ReactComponent as CreditCard } from "../../assets/svg/credit-card.svg";
+// import { ReactComponent as briefcase } from "../../assets/svg/briefcase.svg";
```
Comment (Maianki) [inline src/components/Icon/Icon.tsx:9]: You can remove this commented code.

**Review by Maianki — APPROVED** [review]
Your code looks good I have nothing to suggest.
Things I liked about your code:
- Folder structure
- The proper breaking up of large pieces of code into smaller reusable components.

File: src/components/Box/Box.styled.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  display: ${(p) => p.display || "block"};
+  flex-direction: ${(p) => p.direction || "row"};
+
+  gap: ${(p) => (p.gap ? Size[p.gap] : 0)};
+  justify-content: ${(p) => p.justifyContent || "start"};
+  align-items: ${(p) => p.alignItems || "strech"};
+  align-self: ${(p) => p.alignSelf || "auto"};
+
+  flex-grow: ${(p) => p.grow || 0};
+  flex-shrink: ${(p) => p.shrink || 0};
```
Comment (yash9263) [inline src/components/Box/Box.styled.ts:16]: Not sure if 0 is used intentionally or to set the initial value if it is not then can use `initial` or `unset` for the default values

File: src/components/Button/Button.styled.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  position: relative;
+  cursor: pointer;
+  line-height: ${(p) => (p.size ? ButtonSize[p.size] : ButtonSize.md)};
+
+  ${(props) =>
+    props.disabled &&
+    css`
+      cursor: not-allowed;
+      opacity: 0.5;
+    `}
```
Comment (yash9263) [inline src/components/Button/Button.styled.ts:25]: can use the `button::disabled` pseudo-class itself to style a disabled button. 
```
styled.button{
  &:disabled,
  &[disabled] {
  
  }
}
```

**Review by yash9263 — APPROVED** [review]

**Comment by yash9263** [comment]
Code looks clean and crisp nice folder structure

**Review by sainath7878 — APPROVED** [review]
Everything looked good to me. I have nothing to suggest!

File: src/components/Icon/Icon.tsx (hunk 1 above)
Comment (govindappaarun) [reply src/components/Icon/Icon.tsx:9]: This is required, the thing is eslint was throwing error, so commented it

**Pull request closed without merging**